modules/3d/include/opencv2/3d.hpp

```diff
@@ -2805,6 +2805,61 @@ CV_EXPORTS_W void loadMesh(const String &filename, OutputArray vertices, OutputA
 */
 CV_EXPORTS_W void saveMesh(const String &filename, InputArray vertices, InputArray normals, InputArrayOfArrays indices);
 
+//! Triangle fill settings
+enum class ShadingType
```

Comment:
ShadingType => TriangleShadingType? or make this enumeration nested to TriangleRasterizeSettings

Reply:
My first attempt was to make this enum nested but it lead me to quite bulky code:
```cpp
if (mode == TriangleRasterizeSettings::ShadingType::Flat ||
    mode == TriangleRasterizeSettings::ShadingType::Shaded)
```
I think `TriangleShadingType` would look better. The same about `TriangleCullingMode` and `TriangleRasterizeSettings`.